presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp

```diff
@@ -1132,6 +1132,66 @@ TEST_F(TaskManagerTest, checkBatchSplits) {
   ASSERT_EQ(resultOrFailure.status, nullptr);
 }
 
+TEST_F(TaskManagerTest, spillDirectoryNonExist) {
+  // Cleanup old tasks between test iterations.
+  taskManager_->setOldTaskCleanUpMs(0);
+  for (bool spillDirExist : {false, true}) {
+    SCOPED_TRACE(fmt::format("spillDirExist: {}", spillDirExist));
+    auto spillDir = setupSpillPath();
+
+    const int numBatchesPerFile = 5;
+    const int numFiles = 5;
+    auto filePaths = makeFilePaths(numFiles);
+    std::vector<std::vector<RowVectorPtr>> batches(numFiles);
+    for (int i = 0; i < numFiles; ++i) {
+      batches[i] = makeVectors(numBatchesPerFile, 1'000);
+    }
+    for (int i = 0; i < filePaths.size(); ++i) {
+      writeToFile(filePaths[i]->path, batches[i]);
+    }
+
+    if (spillDirExist) {
+      taskManager_->setBaseSpillDirectory(
+          fmt::format("{}/presto_native/path", spillDir->path));
+    } else {
+      // Set bad spill path.
+      const std::string badSpillPath = "/nonexist/presto_native/path";
+      taskManager_->setBaseSpillDirectory(badSpillPath);
+    }
+
+    std::map<std::string, std::string> queryConfigs;
+    queryConfigs.emplace(core::QueryConfig::kTestingSpillPct, "100");
```

Comment:
Would it make sense to extract a helper function to "enable spilling"?

Reply:
Sure.